src/seedu/addressbook/data/person/UniquePersonList.java

```diff
@@ -84,15 +84,25 @@ public UniquePersonList(UniquePersonList source) {
 
     /**
      * Checks if the list contains an equivalent person as the given argument.
+     * The ReadOnlyPerson.isSamePerson method is used for this comparison, which
+     * defines a weaker notion of equality.
      */
     public boolean contains(ReadOnlyPerson toCheck) {
-        return internalList.contains(toCheck);
+        for (Person p : internalList) {
+            if (p.isSamePerson(toCheck)) {
+                return true;
+            }
+        }
+        return false;
     }
 
     /**
      * Adds a person to the list.
      *
-     * @throws DuplicatePersonException if the person to add is a duplicate of an existing person in the list.
+     * @throws DuplicatePersonException
+     *             if the person to add is a duplicate of an existing person in
+     *             the list. The ReadOnlyPerson.isSamePerson method is used for
+     *             this comparison, which defines a weaker notion of equality.
```

Comment:
Use `{@link package.class#member} ` format to refer to a another method from javadocs